src/components/sbb-teaser-hero/sbb-teaser-hero.tsx

```diff
@@ -1,135 +1,95 @@
-        },
-        mediaQueries: [
-          {
-            conditionFeature: 'max-width',
-            conditionFeatureValue: {
-              lyneDesignToken: true,
-              value: 'sbb-breakpoint-wide-max',
-            },
-            conditionOperator: false,
-          },
-        ],
-      },
-      {
-        image: {
-          height: tokens['sbb-breakpoint-micro-max'],
-          width: tokens['sbb-breakpoint-micro-max'],
-        },
-        mediaQueries: [
-          {
-            conditionFeature: 'max-width',
-            conditionFeatureValue: {
-              lyneDesignToken: true,
-              value: 'sbb-breakpoint-micro-max',
-            },
-            conditionOperator: false,
-          },
-        ],
-      },
-    ],
-  };
+export class SbbTeaserHero implements LinkProperties {
+  /** This will be forwarded as aria-label to anchor tag. */
+  @Prop() public accessibilityLabel: string | undefined;
+
+  /** This will be forwarded as aria-describedby to the anchor tag. */
+  @Prop() public accessibilityDescribedby: string | undefined;
+
+  /** This will be forwarded as aria-labelledby to the anchor tag. */
+  @Prop() public accessibilityLabelledby: string | undefined;
 
-  /**
-   * Text property for sbb-panel. See sbb-panel for additional info
-   */
-  @Prop() public text!: string;
+  /** The href value you want to link to. */
+  @Prop() public href: string | undefined;
 
-  /**
-   * Button text property for sbb-panel. See sbb-panel for additional info
-   */
-  @Prop() public buttonText!: string;
+  /** The relationship of the linked URL as space-separated link types. */
+  @Prop() public rel?: string | undefined;
 
-  /**
-   * Image source property for sbb-image. See sbb-image for additional info
-   */
-  @Prop() public imageSrc!: string;
+  /** Where to display the linked URL. */
+  @Prop() public target?: LinkTargetType | string | undefined;
 
-  /**
-   * Image loading property. See sbb-image for additional info
-   */
-  @Prop() public imageLoading?: InterfaceImageAttributes['loading'] = 'eager';
+  /** Pass in an id, if you need to identify the inner link element. */
+  @Prop() public idValue?: string;
 
-  /**
-   * Link to open if the teaser is clicked/pressed.
-   */
-  @Prop() public link!: string;
+  /** Panel link text. */
+  @Prop() public linkContent?: string;
 
-  /**
-   * If set, the link will be opened in a new window.
-   */
-  @Prop() public openInNewWindow?: boolean;
+  /** Image src will be passed to `sbb-image`. */
+  @Prop() public imageSrc?: string;
 
-  /**
-   * If `openInNewWindow` is set, you should provide according information
-   * which will be read aloud for screenreader users (e.g. "Link target will
-   * open in a new window").
-   */
-  @Prop() public newWindowInfoText?: string;
+  /** Image alt text will be passed to `sbb-image`. */
+  @Prop() public imageAlt?: string;
 
   public render(): JSX.Element {
-    const linkAttributes = {};
+    let TAG_NAME: string;
```

Comment:
I'm not overly fond of this pattern, although it's technically fine.
As a thought, could we extract this to a method?
e.g.
```ts
private resolveRenderVariables() {
  if (this.href) {
    return {
      TAG_NAME: 'a',
      attributes: getLinkAttributeList(this),
      screenReaderNewWindowInfo: !this.accessibilityLabel && this.target === '_blank',
    }
  }
  return { TAG_NAME: 'span', attributes: getLinkButtonBaseAttributeList(this) }
}
...
const { TAG_NAME, attributes, screenReaderNewWindowInfo } = this.resolveRenderVariables();
```

Thoughts?